### DISCUSSION TITLE
Setting `max_concurrent` to `0` in the `forward` plugin.

### DISCUSSION THREAD

**Discussion by ihgann** [coredns/coredns#6969] [Q&A]
Hi,

I'm taking a look at the [forward](https://coredns.io/plugins/forward/) plugin and setting `max_concurrent`. One thing I noticed is that, in practice, setting this value to `0` seems to be the same behavior as not setting it - that is, the maximum concurrent forwarded requests becomes unbounded.

Code confirms this as well: https://github.com/coredns/coredns/blob/master/plugin/forward/forward.go#L98-L105

My question - is this an official contract? It's not stated in any documentation, so I'm curious if this is something that is considered subject to change, or if we can rely on this contract moving forward? It seems to be reasonably stable considering `0` is the "default" for an `int` in Go.

**Comment by chrisohaver** [comment]
Not an official contract. Probably more of an oversight letting users specify 0.

In what context would one rely on this?

**Reply by ihgann** [reply]
My use case is for templating. My initial thought is to always set `max_concurrent`, and in the case of disabling it, I can have the value be 0, instead of omitting the field.

It's not very important either way, I just didn't want to proceed with this thought if it could be considered unstable - it is primarily a question of curiosity. For safety, I'll proceed with omitting the key in this situation.
